### PR TITLE
Adding Kerberos Auth configs from the trino-python-client

### PR DESCRIPTION
**Pull request by BendettaSD**
## Overview
My Kerberos Auth'ed Trino wouldn't connect with dbt without the following parameters from the trino-python-client exposed.
https://github.com/trinodb/trino-python-client/blob/master/trino/auth.py#L42-L53

Original list here.
https://github.com/starburstdata/dbt-trino/blob/master/dbt/adapters/trino/connections.py#L188-L193

Options were exposed through dbt profile variables. 

## Checklist

- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] `README.md` updated and added information about my change
- [x] I have run `changie new` to [create a changelog entry](https://github.com/starburstdata/dbt-trino/blob/master/CONTRIBUTING.md#Adding-CHANGELOG-Entry)

### CODE REVIEW

**Comment by BendettaSD** [comment]
Happy to take any feedback. I wasn't sure of a good way to test complicated Kerberos setups so I didn't add any further tests.

File: .changes/unreleased/Fixes-20220926-091442.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+kind: Fixes
+body: Adding Kerberos Auth configsfrom the trino-python-client
```
Comment (hovaesco) [inline .changes/unreleased/Fixes-20220926-091442.yaml:2]:
```suggestion
body: Adding Kerberos auth configs from the trino-python-client
```

**Review by hovaesco — COMMENTED** [review]
Looks good, please squash commits.

File: README.md [hunk 2]
```diff
@@ -41,26 +41,31 @@ A dbt profile can be configured to run against Trino using the following configu
...
 | database                       | Specify the database to build models into                                                                    | Required                                                                                                         | `analytics`                      |
 | schema                         | Specify the schema to build models into. Note: it is not recommended to use upper or mixed case schema names | Required                                                                                                         | `public`                         |
-| host                           | The hostname to connect to                                                                                   | Required                                                                                                         | `127.0.0.1`                      |
+| host                           | The hostname to connect to                                                                                   | Required                                                                                                         | `127.0.0.1`                               |
```
Comment (findinpath) [inline README.md:65]: nit: make sure to have the delimiters of the columns of the table aligned

File: README.md [hunk 3]
```diff
@@ -41,26 +41,31 @@ A dbt profile can be configured to run against Trino using the following configu
...
-| client_private_key             | Path to client private key to be used for certificate based authentication                                   | Optional (required if `method` is `certificate`)                                                                 | `/tmp/tls.key`                   |
+| mutual_authentication          | Boolean flag for mutual authentication                                                                       | Optional (may be required if `method` is `kerberos`)                                                             | `false`                          |
+| force_preemptive               | Boolean flag for preemptively initiate the Kerberos GSS exchange                                             | Optional (may be required if `method` is `kerberos`)                                                             | `false`                          |
+| hostname_override              | Kerberos hostname for a host whose DNS name doesn't match                                                    | Optional (may be required if `method` is `kerberos`)                                                             | `EXAMPLE.COM`                             |
```
Comment (findinpath) [inline README.md:53]: in case there is additional aligning needed in the table, please do it in a separate commit to easily distinguish what has been changed for kerberos purposes.

**Comment by findinpath** [comment]
squash commits that relate to kerberos

**Comment by BendettaSD** [comment]
Thanks for the quick review and notes. I'll clean it up!

**Comment by BendettaSD** [comment]
Cleaned up the whitespaces and squashed commits.

File: .changes/unreleased/Fixes-20220926-091442.yaml (hunk 1 above)
Comment (BendettaSD) [reply .changes/unreleased/Fixes-20220926-091442.yaml]: Edited

File: README.md (hunk 2 above)
Comment (BendettaSD) [reply README.md]: Removed formatting changes in non-added lines.

**Review by hovaesco — APPROVED** [review]
